WalletWasabi.Tests/UnitTests/Tor/TorBinaryHashesTests.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Threading;
+using WalletWasabi.Microservices;
+using Xunit;
+
+namespace WalletWasabi.Tests.UnitTests.Tor
+{
+	public class TorBinaryHashesTests
+	{
+		[Fact]
+		public void VerifyTorBinaryChecksumHashes()
+		{
+			using var cts = new CancellationTokenSource(5_000);
+
+			Dictionary<OSPlatform, string> expectedHashes = new()
+			{
+				{ OSPlatform.Windows, "d244a89f7ca9da2259925affa0bd008d3cdea1d9dfd24cf53efffb7c1eadc169" },
+				{ OSPlatform.Linux, "af7302d62fc1e47f79af8860541365f77547233404302a1e601e1f367e6e2888" },
+				{ OSPlatform.OSX, "fe6d719e18bf3a963f0274de259b7e029f40e4fe778f4d170bba343eb491af00" },
+			};
+
+			foreach (var item in expectedHashes)
+			{
+				string binaryFolder = Path.Combine(MicroserviceHelpers.GetBinaryFolder(item.Key), "Tor");
+				string filePath = Path.Combine(binaryFolder, item.Key == OSPlatform.Windows ? "tor.exe" : item.Key == OSPlatform.Linux ? "tor" : "tor.real");
```

Comment:
We should have a function that returns `TorBinaryPath`. Similar code is already in `TorSettings`.

Reply:
How is it done when starting Tor? Somewhere at TorProcessManager @yahiheb can you check it?

Reply:
I would do something along these lines: https://github.com/kiminuo/WalletWasabi/commit/11161d7dde664aba6b5ecc6093c4dd094cb0d833

Reply:
Should we add this in this PR or in another one?

Reply:
I think you can add it to this PR.

Reply:
Can you try to use this? 

https://github.com/molnard/WalletWasabi/blob/934593a254f622bbab955f0b24e8b8dec309b6f8/WalletWasabi/Tor/TorSettings.cs#L19

Reply:
I cherry picked @kiminuo commit.